croptalk/model_openai_functions.py

```diff
@@ -101,26 +101,37 @@ def get_model(self) -> Tuple[Chain, AgentTokenBufferMemory]:
             | llm_with_tools
             | OpenAIFunctionsAgentOutputParser()
         )
-        memory_llm = ChatOpenAI(model=self.llm_model_name, temperature=0)
-        memory = AgentTokenBufferMemory(
-            memory_key=self.memory_key,
-            llm=memory_llm,
-            max_token_limit=6000,
-        )
 
-        agent_executor = (
-            AgentExecutor(
+        if no_memory:
+            memory = None
+        else:
+            memory_llm = ChatOpenAI(model=self.llm_model_name, temperature=0)
+            memory = AgentTokenBufferMemory(
+                memory_key=self.memory_key,
+                llm=memory_llm,
+                max_token_limit=6000,
+            )
+
+        agent_executor = AgentExecutor(
                 agent=agent,
                 tools=tools,
                 memory=memory,
                 verbose=True,
                 max_iterations=10,
-                return_intermediate_steps=True,
-            )
+                return_intermediate_steps=True, # we always want to return the intermediate steps to see them in the run trace
```

Comment:
just a thought : may this affect speed performance?

Reply:
Not sure, but good question... we may want to test that at some point